lib/Geocode.php

```diff
@@ -1612,7 +1618,7 @@ function lookup()
 					if ($this->bIncludePolygonAsKML) $sSQL .= ",ST_AsKML(geometry) as askml";
 					if ($this->bIncludePolygonAsSVG) $sSQL .= ",ST_AsSVG(geometry) as assvg";
 					if ($this->bIncludePolygonAsText || $this->bIncludePolygonAsPoints) $sSQL .= ",ST_AsText(geometry) as astext";
-					$sSQL .= " from placex where place_id = ".$aResult['place_id'];
+					$sSQL .= " from (select place_id,centroid,ST_SimplifyPreserveTopology(geometry,".$this->fPolygonSimplificationThreshold.") as geometry from placex where place_id = ".$aResult['place_id'].") as plx";
```

Comment:
On Tue, Apr 28, 2015 at 9:38 PM, Sarah Hoffmann <notifications@github.com>
wrote:

> In lib/Geocode.php
> <https://github.com/twain47/Nominatim/pull/267#discussion_r29280174>:
>
> > @@ -1612,7 +1618,7 @@ function lookup()
> >  					if ($this->bIncludePolygonAsKML) $sSQL .= ",ST_AsKML(geometry) as askml";
> >  					if ($this->bIncludePolygonAsSVG) $sSQL .= ",ST_AsSVG(geometry) as assvg";
> >  					if ($this->bIncludePolygonAsText || $this->bIncludePolygonAsPoints) $sSQL .= ",ST_AsText(geometry) as astext";
> > -					$sSQL .= " from placex where place_id = ".$aResult['place_id'];
> > +					$sSQL .= " from (select place_id,centroid,ST_SimplifyPreserveTopology(geometry,".$this->fPolygonSimplificationThreshold.") as geometry from placex where place_id = ".$aResult['place_id'].") as plx";
>
> I wonder if we should preserve the original SQL here for the frequent case
> that polygon_threshold = 0.0.
>
Well, we totally could, but that would make the code a little bit more
complicated.  And I'm not sure if x == 0.0 works all that great with
floating point?  For that we could also set the default value to -1, and
check for < 0.